lib/reddex_web/plugs/auth.ex

```diff
@@ -0,0 +1,21 @@
+defmodule ReddexWeb.Plugs.Auth do
+  @moduledoc "Check if current_user is present in the session"
+
+  import Plug.Conn
+
+  def init(opts), do: opts
+
+  def call(conn, _) do
+    # TODO: Not sure about this
+    # maybe be more explicit in controllers with actions
+    if get_session(conn, :current_user) || conn.request_path == "/sign_in" do
+      conn
+    else
+      redirect_to_login(conn)
+    end
+  end
+
+  defp redirect_to_login(conn) do
+    conn |> Phoenix.Controller.redirect(to: "/sign_in") |> halt()
```

Comment:
```
conn
|> Phoenix.Controller.redirect(to: "/sign_in")
|> halt()
``` ?

Reply:
i'm not sure, i can change but function name and length should be clear enough from my point of view